extensions/hibernate-validator/runtime/src/main/java/io/quarkus/hibernate/validator/runtime/HibernateValidatorRecorder.java

```diff
@@ -150,14 +151,23 @@ public void created(BeanContainer container) {
                     configuration.addValueExtractor(valueExtractor);
                 }
 
-                List<InstanceHandle<HibernateValidatorFactoryCustomizer>> constraintMappingProviderList = Arc.container()
+                List<InstanceHandle<HibernateValidatorFactoryCustomizer>> hibernateValidatorFactoryCustomizers = Arc.container()
                         .listAll(HibernateValidatorFactoryCustomizer.class);
-                for (InstanceHandle<HibernateValidatorFactoryCustomizer> cmpInstanceHandle : constraintMappingProviderList) {
-                    if (cmpInstanceHandle.isAvailable()) {
-                        final HibernateValidatorFactoryCustomizer hibernateValidatorFactoryCustomizer = cmpInstanceHandle.get();
+                for (InstanceHandle<HibernateValidatorFactoryCustomizer> hibernateValidatorFactoryInstanceHandle : hibernateValidatorFactoryCustomizers) {
+                    if (hibernateValidatorFactoryInstanceHandle.isAvailable()) {
+                        final HibernateValidatorFactoryCustomizer hibernateValidatorFactoryCustomizer = hibernateValidatorFactoryInstanceHandle
+                                .get();
                         hibernateValidatorFactoryCustomizer.customize(configuration);
                     }
                 }
+                List<InstanceHandle<ValidatorFactoryCustomizer>> validatorFactoryCustomizers = Arc.container()
+                        .listAll(ValidatorFactoryCustomizer.class);
+                for (InstanceHandle<ValidatorFactoryCustomizer> validatorFactoryInstanceHandle : validatorFactoryCustomizers) {
+                    if (validatorFactoryInstanceHandle.isAvailable()) {
+                        final ValidatorFactoryCustomizer validatorFactoryCustomizer = validatorFactoryInstanceHandle.get();
+                        validatorFactoryCustomizer.customize(configuration);
+                    }
+                }
```

Comment:
You talk about ordering the in the documentation and about using `@Priority`, but this code will apply `HibernateValidatorFactoryCustomizer` first, then `ValidatorFactoryCustomizer`, regardless of priority.

If you're going to do the renaming after all, I'd suggest the simpler option of having `HibernateValidatorFactoryCustomizer` extend `ValidatorFactoryCustomizer`?

Reply:
That was my first intent, but then I thought it would be messy, I don't know why. Anyway, done, and indeed much better.